### PR TITLE
Add SimpleHttpProvider for GraphClientFactory compatibility

### PR DESCRIPTION
**Pull request by andrueastman**
### Changes proposed in this pull request
- Adds a new IHttpProvider to enhance compatibility with GraphClientFactory by working towards adding support for the following scenario.
```
  HttpClient client = GraphClientFactory.Create(...);
  var graphServiceClient = new GraphServiceClient(client);

```
This means that one can easily use a custom httpClient with the request builders provided.
- Modifies checks in BaseRequests.cs to not authenticate the requests when the Provider is 
 either SimpleHttprovider or HttpProvider  and when authprovider is present.
- Adds tests

### Other links
- resolves a task in #27  
- is the predecessor for https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/221

### CODE REVIEW

File: src/Microsoft.Graph.Core/Microsoft.Graph.Core.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <DelaySign>false</DelaySign>
     <AssemblyOriginatorKeyFile>35MSSharedLib1024.snk</AssemblyOriginatorKeyFile>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <VersionPrefix>1.17.0</VersionPrefix>
+    <VersionPrefix>1.18.0</VersionPrefix>
     <VersionSuffix></VersionSuffix>
     <PackageReleaseNotes>
       - Removed PageIterator. It is now available from the service library.
```
Comment (MIchaelMainer) [inline src/Microsoft.Graph.Core/Microsoft.Graph.Core.csproj]: Remove old release notes. Add a release note for this work. Will be visible from NuGet.org and the package.

File: src/Microsoft.Graph.Core/Requests/SimpleHttpProvider.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                        if (response.Headers.TryGetValues(CoreConstants.Headers.ThrowSiteHeaderName, out var throwSiteValues))
+                        {
+                            error.ThrowSite = throwSiteValues.FirstOrDefault();
+                        }
+                    }
+
+                    // Pass through the response headers and status code to the ServiceException.
+                    // System.Net.HttpStatusCode does not support RFC 6585, Additional HTTP Status Codes.
+                    // Throttling status code 429 is in RFC 6586. The status code 429 will be passed through.
+                    throw new ServiceException(error, response.Headers, response.StatusCode);
```
Comment (MIchaelMainer) [inline src/Microsoft.Graph.Core/Requests/SimpleHttpProvider.cs]: Please review #37 as it adds some additional functionality to the HttpProvider that this provider should use to better support error/diagnostic info. Actually, I need a reviewer :). Let's pull 37 in and then we can pull yours in.

File: src/Microsoft.Graph.Core/Microsoft.Graph.Core.csproj (hunk 1 above)
Comment (andrueastman) [reply src/Microsoft.Graph.Core/Microsoft.Graph.Core.csproj]: Resolved

File: src/Microsoft.Graph.Core/Requests/SimpleHttpProvider.cs (hunk 2 above)
Comment (andrueastman) [reply src/Microsoft.Graph.Core/Requests/SimpleHttpProvider.cs]: That sounds like a plan.

**Comment by andrueastman** [comment]
We will pull this in once #37 is completed and I have pull the necessary error/diagnostic changes

**Comment by andrueastman** [comment]
New support for error/diagnostic info has now been incorporated. This PR should now be ready to be pulled in.

@MIchaelMainer @darrelmiller @irvinesunday

**Comment by MIchaelMainer** [comment]
It looks fine - just set the version back to 1.18.0 and then we should be set. Part of me wants to turn some of those thrown ServiceException into ClientException as they aren't really ServiceException. Perhaps we do that wholesale at a later break.

**Comment by andrueastman** [comment]
Version number has now been corrected to 1.18.0

**Comment by darrelmiller** [comment]
Looks like merging the two branches broke some tests.  Seems related to ThrowSite. /cc @andrueastman  @MIchaelMainer

**Comment by andrueastman** [comment]
Thanks pointing this out. This has been resolved. 

To prevent this from occurring again, I think we might need to look into the pr validation checks as it looks to me like the tests are not really run.

**Pull request merged**